pages/starsearch.tsx

```diff
@@ -24,28 +24,107 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     return { notFound: true };
   }
 
-  return { props: { userId } };
+  return { props: { userId, bearerToken: session?.access_token } };
 }
 
 type StarSearchPageProps = {
   userId: number;
+  bearerToken: string;
 };
 
-export default function StarSearchPage({ userId }: StarSearchPageProps) {
+type StarSearchChat = {
+  author: "You" | "StarSearch";
+  content: string;
+};
+
+export default function StarSearchPage({ userId, bearerToken }: StarSearchPageProps) {
   const [starSearchState, setStarSearchState] = useState<"initial" | "chat">("initial");
+  const [chat, setChat] = useState<StarSearchChat[]>([]);
```

Comment:
One thing to consider potentially down the line @jpmcb is do we store past coversations like ChatGPT or similar services? If so, I imagine that adds to GDPR?

Reply:
We need a data export function and updated retention and data use policy.

That is something on my list. But something we need for a public launch.

Reply:
We currently do not store conversations or the input query. But, yes, in the future if we do store those, we'd need to come up with some strategy to get rid of those. Probably just a table in the database that drops them after 1 year or so.